depdive/src/update.rs

```diff
@@ -294,49 +345,97 @@ impl UpdateAnalyzer {
 
             let stats = diff.stats()?;
 
+            let modified_build_scripts = dep_change_info.build_script_paths.as_ref().map(|paths| {
+                HashSet::from_iter(
```

Comment:
Use `collect` instead of `from_iter` as it's a little more idiomatic. If you need type hints because inference isn't working you can use `.collect<HashSet<_>>()`.